docs/source/deployment_client.rst

```diff
@@ -0,0 +1,119 @@
+
+
+Builtin Deployment Clients
+--------------------------
+
+The MLflow library provides several built-in deployment clients that can be used to interact with models served on different platforms. The following are the built-in deployment clients provided by MLflow:
+
+* :py:class:`mlflow.deployments.MlflowDeploymentClient`: A deployment client for served models in the MLflow Deployments Server
+* :py:class:`mlflow.deployments.DatabricksDeploymentClient`: A deployment client for served models on Databricks.
+* :py:class:`mlflow.deployments.OpenAIDeploymentClient`: A deployment client for served models on OpenAI or Azure OpenAI platforms.
+
+Deployment Client Factory functions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+These clients can be accessed using the following factory methods:
+
+* :py:func:`mlflow.deployments.get_deploy_client`: Returns an instance of the appropriate deployment client based on the specified deployment target.
+* :py:func:`mlflow.deployments.get_deployments_target`: Returns the deployment target associated with the specified deployment client.
+* :py:func:`mlflow.deployments.set_deployments_target`: Sets the deployment target for the specified deployment client. The deployment target can also be set using the ``MLFLOW_DEPLOYMENT_TARGET`` environment variable. If running in Databricks, the deployments target will be set to ``databricks`` by default.
+
+The following is an example of how to create an instance of a Databricks deployment client:
+
+.. code-block:: python
+
+    from mlflow.deployments import get_deploy_client
+
+    client = get_deploy_client("databricks")
+
+Custom Deployment Clients
+-------------------------
+
+MLflow also provides a :py:class:`mlflow.deployments.BaseDeploymentClient` interface that can be used to implement custom deployment clients. These custom deployment clients can be installed as plugins. For more information, see `MLflow Plugins <https://mlflow.org/docs/latest/plugins.html>`_.
+
+Deployment Client Query APIs
+----------------------------
+
+MLflow deployment clients provide a unified query API for deployments or model endpoints. This is the :py:func:`mlflow.deployments.BaseDeploymentClient.predict` API.
+
+For example, to query ``gpt-4`` model endpoint on OpenAI, you can use the following code:
+
+.. code-block:: python
+
+    import os
+    from mlflow.deployments import get_deploy_client
+
+    os.environ["OPENAI_API_KEY"] = "YOUR_API_KEY"
+
+    client = get_deploy_client("openai")
+    client.predict(
+        endpoint="gpt-4",
+        inputs={
+            "messages": [
+                {"role": "user", "content": "Hello!"},
+            ],
+        },
+    )
+
+Deployment Client CRUD APIs
+---------------------------
+
+For platforms that support CRUD operations on endpoints, MLflow provides APIs to create, update, delete, and list endpoints. These APIs are available in the :py:class:`mlflow.deployments.BaseDeploymentClient` interface.
```

Comment:
Could we list a few examples of endpoint providers that support CRUD and those that don't (this might be a frequently asked question for readers)